src/main/java/org/cryptomator/ui/mainwindow/ResizeController.java

```diff
@@ -1,41 +1,100 @@
+	public Region bDefaultRegion;
 
 	private double origX, origY, origW, origH;
 
+	private final Settings settings;
+
+	private final BooleanBinding showResizingArrows;
+
 	@Inject
-	ResizeController(@MainWindow Stage window) {
+	ResizeController(@MainWindow Stage window, Settings settings) {
 		this.window = window;
-		// TODO inject settings and save current position and size
+		this.settings = settings;
+		this.showResizingArrows = Bindings.createBooleanBinding(this::isShowResizingArrows, window.fullScreenProperty());
 	}
 
 	@FXML
 	public void initialize() {
-		tlResizer.setOnMousePressed(this::startResize);
-		trResizer.setOnMousePressed(this::startResize);
-		blResizer.setOnMousePressed(this::startResize);
-		brResizer.setOnMousePressed(this::startResize);
-		tlResizer.setOnMouseDragged(this::resizeTopLeft);
-		trResizer.setOnMouseDragged(this::resizeTopRight);
-		blResizer.setOnMouseDragged(this::resizeBottomLeft);
-		brResizer.setOnMouseDragged(this::resizeBottomRight);
+		LOG.trace("init ResizeController");
+
+		if (neverTouched()) {
+			settings.displayConfigurationProperty().setValue(getMonitorSizes());
+			return;
+		} else {
+			if (didDisplayConfigurationChange()) {
+				//If the position is illegal, then the window appears on the main screen in the middle of the window.
+				Rectangle2D primaryScreenBounds = Screen.getPrimary().getBounds();
+				window.setX((primaryScreenBounds.getWidth() - window.getMinWidth()) / 2);
+				window.setY((primaryScreenBounds.getHeight() - window.getMinHeight()) / 2);
+				window.setWidth(window.getMinWidth());
+				window.setHeight(window.getMinHeight());
+			} else {
+				window.setHeight(settings.windowHeightProperty().get() > window.getMinHeight() ? settings.windowHeightProperty().get() : window.getMinHeight());
+				window.setWidth(settings.windowWidthProperty().get() > window.getMinWidth() ? settings.windowWidthProperty().get() : window.getMinWidth());
+				window.setX(settings.windowXPositionProperty().get());
+				window.setY(settings.windowYPositionProperty().get());
+			}
+		}
+		savePositionalSettings();
+	}
+
+	private boolean neverTouched() {
+		return (settings.windowHeightProperty().get() == 0) && (settings.windowWidthProperty().get() == 0) && (settings.windowXPositionProperty().get() == 0) && (settings.windowYPositionProperty().get() == 0);
+	}
+
+	private boolean didDisplayConfigurationChange() {
+		String currentDisplayConfiguration = getMonitorSizes();
+		String settingsDisplayConfiguration = settings.displayConfigurationProperty().get();
+		boolean configurationHasChanged = !settingsDisplayConfiguration.equals(currentDisplayConfiguration);
+		if (configurationHasChanged) settings.displayConfigurationProperty().setValue(currentDisplayConfiguration);
+		return configurationHasChanged;
+	}
+
+	private String getMonitorSizes() {
+		ObservableList<Screen> screens = Screen.getScreens();
+		StringBuilder sb = new StringBuilder();
+		for (int i = 0; i < screens.size(); i++) {
```

Comment:
Instead of iterating over all Screens, you can just use the primary one, see https://openjfx.io/javadoc/16/javafx.graphics/javafx/stage/Screen.html#getPrimary().

As long as this one does not change (or better its resolution), we do not have to change anything.

Reply:
If the windows was located on a secondary screen before it got disconnected, we better recognize this change. Therefore we need to remember the resolution of all screens.